### PR TITLE
Promisify inject if needed

### PR DESCRIPTION
**Pull request by allevo**
The implementation is not so good.

Any advice?

### CODE REVIEW

File: test/inject.test.js [hunk 1]
```diff
@@ -222,6 +222,72 @@ test('inject get request - reply stream', t => {
...
+    reply.send(payload)
+  })
+
+  fastify.inject({
+    method: 'GET',
+    url: '/'
+  })
+    .then(res => {
+      t.strictEqual(res.statusCode, 200)
+    })
```
Comment (mcollina) [inline test/inject.test.js]: is this passing linting? can you format this better?

Comment (allevo) [reply test/inject.test.js]: Yeah, travis says that.

> can you format this better?

😃

File: fastify.js [hunk 2]
```diff
@@ -466,18 +466,40 @@ function build (options) {
...
 
     if (started) {
-      shot.inject(this, opts, cb)
+      return _inject(this, opts, cb)
+    }
+
+    if (cb) {
+      this.ready(err => {
+        if (err) throw err
+        return _inject(this, opts, cb)
```
Comment (mcollina) [inline fastify.js]: I think you can avoid defining `_inject()`  and just use `shot.inject` here.

File: test/inject.test.js (hunk 1 above)
Comment (mcollina) [reply test/inject.test.js]: this is still not nice to read

File: fastify.js (hunk 2 above)
Comment (allevo) [reply fastify.js]: Yes, but with `_inject` there's encapsulated the "`shot` logic".

Anyway if you like more the substitution, I'll do it!

Comment (mcollina) [reply fastify.js]: Yes, it would be less code overall.

File: fastify.js [hunk 3]
```diff
@@ -466,18 +466,40 @@ function build (options) {
...
 
+  function _inject (server, opts, cb) {
+    if (cb) {
+      shot.inject(server, opts, cb)
+      return
+    }
+    return new Promise(resolve => {
+      shot.inject(server, opts, resolve)
+    })
+  }
```
Comment (mcollina) [inline fastify.js]: `_inject` is still defined here, can you remove the full function and inline calls to `shot.inject`  in `inject`?

**Pull request merged**